### PR TITLE
Precompile stdlib images and include them in sysimg

### LINKED ISSUE #24697: Changes to stdlib (Test) don't cause recompilation of sysimage
Currently `Test` is still part of the sysimage, but changing the files does not trigger a rebuild of it.

### LINKED ISSUE #24765: stdlib packages missing from precompiled sysimg
All of the files in stdlib need to declare `__precompile__()` and should probably also be baked into the sysimg currently. (See the "strawman example" https://github.com/JuliaLang/julia/pull/23876/files#diff-9a4126dd8386dbc7e2eb93f7bef02d1bR457 for how this needs to be done.)

### PR DESCRIPTION
**Pull request by vchuravy**
* Marks all of stdlib as `__precompile__(true)`
* Includes all of stdlib in `base\sysimg.jl`
* Ensures that sysimg get's rebuild when stlib files change

fixes #24765
fixes #24697

### CODE REVIEW

**Comment by ViralBShah** [comment]
Thank you. I wasn't clear what all the stdlib incantations were.

File: stdlib/CRC32c/src/CRC32c.jl [hunk 1]
```diff
@@ -1,3 +1,6 @@
+
```
Comment (vchuravy) [inline stdlib/CRC32c/src/CRC32c.jl]: Probably should have a license header here.

File: Makefile [hunk 2]
```diff
@@ -199,6 +199,7 @@ CORE_SRCS := $(addprefix $(JULIAHOME)/, \
 		base/reflection.jl \
 		base/tuple.jl)
 BASE_SRCS := $(sort $(shell find $(JULIAHOME)/base -name \*.jl) $(shell find $(BUILDROOT)/base -name \*.jl))
+STDLIB_SRCS := $(sort $(shell find $(JULIAHOME)/stdlib/*/src -name \*.jl) $(shell find $(BUILDROOT)/stdlib/*/src -name \*.jl))
```
Comment (vtjnash) [inline Makefile]: I don't think buildroot should have a stdlib folder

**Review by vtjnash — APPROVED** [review]

**Comment by vchuravy** [comment]
Will merge later today if there are no objections.

**Pull request merged**